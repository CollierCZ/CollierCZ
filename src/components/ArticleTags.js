import React from "react";
import { navigate } from "gatsby";
import { arrayOf, string } from "prop-types";
import { slugify } from "../utilities/CaseHelpers";
import { TextLink } from "@kiwicom/orbit-components";

const ArticleTags = ({ prefix, tags }) => {
  return (
    <span>
      {prefix}
      {tags.map((tag, index, arr) => (
        <span key={tag}>
          {" "}
          <TextLink
            href={`/tags/${slugify(tag)}`}
            type="secondary"
            onClick={(event) => {
              event.preventDefault();
              navigate(`/tags/${slugify(tag)}`);
            }}
            stopPropagation
          >
            {tag}
          </TextLink>
          {index !== arr.length - 1 ? ", " : ""}
        </span>
      ))}
    </span>
  );
};

ArticleTags.propTypes = {
  prefix: string,
  tags: arrayOf(string.isRequired).isRequired,
};

export default ArticleTags;
