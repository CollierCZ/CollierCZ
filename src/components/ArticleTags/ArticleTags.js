import React, { Component } from "react";
import _ from "lodash";
import { Link } from "gatsby";
import "./ArticleTags.css";

class ArticleTags extends Component {
  render() {
    const { prefix, tags } = this.props;
    if (tags) {
      let tagArray = tags.split(",");
      return (
                <span>
          {prefix}
          {tagArray.map((tag, index, arr) => (
            <span key={tag}>
              <Link key={tag} to={`/tags/${_.kebabCase(tag)}`}>
                {tag}
              </Link>
              {index !== arr.length - 1 ? ", " : ""}
            </span>
          ))}
        </span>
      );
    }
    return null;
  }
}

export default ArticleTags;
