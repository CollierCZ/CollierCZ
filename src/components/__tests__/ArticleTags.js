import React from "react"
import { shallow } from "enzyme"
import ArticleTags from "../ArticleTags"

const tags = ["documentation","UX","experience mapping"]

describe("ArticleTags", () => {
  it("renders tags correctly", () => {
    const component = shallow (<ArticleTags tags={tags} prefix=" on " />);
    expect(component).toMatchSnapshot()
  })
  it("renders correctly when empty", () => {
    const component = shallow (<ArticleTags prefix=" on " />);
    expect(component).toMatchSnapshot()
  })
})