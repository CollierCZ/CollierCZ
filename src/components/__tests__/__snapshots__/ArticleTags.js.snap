// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTags renders tags correctly 1`] = `
.c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c0,
.c0:link,
.c0:visited {
  color: #252A31;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c0:hover,
.c0:active,
.c0:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<span>
   on 
  <span>
     
    <a
      className="c0"
      href="/tags/documentation"
      onClick={[Function]}
    >
      documentation
    </a>
    , 
  </span>
  <span>
     
    <a
      className="c0"
      href="/tags/ux"
      onClick={[Function]}
    >
      UX
    </a>
    , 
  </span>
  <span>
     
    <a
      className="c0"
      href="/tags/experience-mapping"
      onClick={[Function]}
    >
      experience mapping
    </a>
    
  </span>
</span>
`;
